### PR TITLE
Connector hive 

### PR DESCRIPTION
**Pull request by zhljen**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -23,3 +23,4 @@ slf4j_version=1.7.12
 swagger_version=1.3.12
 hive_version=1.2.1
 hadoopcore_version=1.2.1
+hivemetastore_version=1.2.1
```
Comment (ajoymajumdar) [inline gradle.properties]: Can we use the hive_version property? This way we use a consistent version of hive modules.

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorDatabaseService.java [hunk 2]
```diff
@@ -65,8 +76,12 @@ public HiveConnectorDatabaseService(@Nonnull final MetacatHiveClient metacatHive
     public void create(@Nonnull final ConnectorContext requestContext, @Nonnull final DatabaseInfo databaseInfo) {
         try {
             this.metacatHiveClient.createDatabase(hiveMetacatConverters.fromDatabaseInfo(databaseInfo));
-        } catch (TException exception) {
+        } catch (AlreadyExistsException exception) {
             throw new DatabaseAlreadyExistsException(databaseInfo.getName(), exception);
+        } catch (MetaException exception) {
+            throw new InvalidMetaException(databaseInfo.getName(), exception);
+        } catch (TException exception) {
```
Comment (ajoymajumdar) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorDatabaseService.java]: Replace TException with ConnectorException.

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFactory.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        this.name = name;
+        this.configuration = configuration;
+        this.infoConverter = infoConverter;
+        init();
+    }
+
+    private void init() {
+        //JPA module
+        final Map<String, Object> props = Maps.newHashMap(configuration);
+        props.put("hibernate.connection.datasource",
```
Comment (ajoymajumdar) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFactory.java]: Why do you need this?

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorPlugin.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Override
+    public String getType() {
+        return CONNECTOR_TYPE;
+    }
+
+    @Override
+    public ConnectorFactory create(@Nonnull final String connectorName,
+                                   @Nonnull final Map<String, String> configuration) {
+        return new HiveConnectorFactory(connectorName, configuration, (HiveConnectorInfoConverter) getInfoConverter());
```
Comment (ajoymajumdar) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorPlugin.java]: Can we use the INO_CONVERTER_HIVE?

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/converters/HiveConnectorInfoConverter.java [hunk 5]
```diff
@@ -76,7 +83,8 @@ public DatabaseInfo toDatabaseInfo(final QualifiedName qualifiedName, final Data
     public Database fromDatabaseInfo(final DatabaseInfo databaseInfo) {
         final QualifiedName databaseName = databaseInfo.getName();
         final String name = (databaseName == null) ? "" : databaseName.getDatabaseName();
-        final String dbUri = Strings.isNullOrEmpty(databaseInfo.getUri()) ? "" : databaseInfo.getUri();
+        //this is a temp hack to resolve the uri = null issue
+        final String dbUri = Strings.isNullOrEmpty(databaseInfo.getUri()) ? "file://temp/" : databaseInfo.getUri();
```
Comment (ajoymajumdar) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/converters/HiveConnectorInfoConverter.java]: Uri is set by the metastore by default if the uri is null. So we should just pass whatever is set by the user.

File: gradle.properties (hunk 1 above)
Comment (zhljen) [inline gradle.properties]: yes.

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFactory.java (hunk 3 above)
Comment (zhljen) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorFactory.java]: Not needed, will remove

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorPlugin.java (hunk 4 above)
Comment (zhljen) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorPlugin.java]: sure

File: metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/converters/HiveConnectorInfoConverter.java (hunk 5 above)
Comment (zhljen) [inline metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/converters/HiveConnectorInfoConverter.java]: Looked at the current code, we always set this as ""
MapStructHiveConverters.java line 139 - 142
        String dbUri = dto.getUri();
        if (Strings.isNullOrEmpty(dbUri)) {
            dbUri = "";
        }
        database.setLocationUri(dbUri);

**Pull request merged**